src/base/static/components/place-detail-view/place-detail-survey.js

```diff
@@ -0,0 +1,192 @@
+// TODO: localization in this component.
+
+import React, { Component } from "react";
+import PropTypes from "prop-types";
+import {
+  Map as ImmutableMap,
+  OrderedMap as ImmutableOrderedMap,
+} from "immutable";
+import emitter from "../utils/emitter";
+
+import FormField from "../form-field";
+import SecondaryButton from "../ui-elements/secondary-button";
+import PlaceDetailSurveyResponse from "./place-detail-survey-response";
+import WarningMessagesContainer from "../ui-elements/warning-messages-container";
+import Avatar from "../ui-elements/avatar";
+
+import constants from "../constants";
+import { placeDetailSurvey as messages } from "../messages";
+
+import "./place-detail-survey.scss";
+
+class PlaceDetailSurvey extends Component {
+  constructor(props) {
+    super(props);
+    let fields = ImmutableOrderedMap();
+    this.props.surveyConfig.items.forEach(field => {
+      fields = fields.set(
+        field.name,
+        ImmutableMap()
+          .set(constants.FIELD_STATE_RENDER_KEY, Math.random())
```

Comment:
Why do we still need this render key? Is this VV-related?

Reply:
We use them here for the same reason we use the in the VV form-- this survey form can be "restarted" (i.e. users can submit multiple comments from the same form).